src/renderer/canvas/label.ts

```diff
@@ -58,40 +72,160 @@ export class Label {
   }
 }
 
-export const drawLabel = (context: CanvasRenderingContext2D, label: Label) => {
+export const drawLabel = (context: CanvasRenderingContext2D, label: Label, type: 'node' | 'edge') => {
   const isDrawable = label.textLines.length > 0 && label.fontSize > 0;
   if (!isDrawable || !label.position) {
     return;
   }
 
-  drawTextBackground(context, label);
-  drawText(context, label);
+  const borderPoints: BorderPoint[] | undefined = drawTextBackgroundAndBorder(context, label, type);
+
+  if (!borderPoints) {
+    return;
+  }
+
+  drawTextBackground(context, borderPoints, label);
+  drawTextBackgroundBorder(context, borderPoints, label);
+  drawText(context, borderPoints, label, type);
 };
 
-const drawTextBackground = (context: CanvasRenderingContext2D, label: Label) => {
-  if (!label.properties.fontBackgroundColor || !label.position) {
+const drawTextBackgroundAndBorder = (
+  context: CanvasRenderingContext2D,
+  label: Label,
+  type: 'node' | 'edge',
+): BorderPoint[] | undefined => {
+  if (!label.position) {
     return;
   }
 
-  context.fillStyle = label.properties.fontBackgroundColor.toString();
   const margin = label.fontSize * FONT_BACKGROUND_MARGIN;
-  const height = label.fontSize + 2 * margin;
+
   const lineHeight = label.fontSize * FONT_LINE_SPACING;
+
   const baselineHeight = label.textBaseline === LabelTextBaseline.MIDDLE ? label.fontSize / 2 : 0;
 
+  const borderPoints: BorderPoint[] = [];
+
   for (let i = 0; i < label.textLines.length; i++) {
     const line = label.textLines[i];
-    const width = context.measureText(line).width + 2 * margin;
-    context.fillRect(
-      label.position.x - width / 2,
-      label.position.y - baselineHeight - margin + i * lineHeight,
-      width,
-      height,
-    );
+
+    let width = context.measureText(line).width + 2 * margin;
```

Comment:
Btw there is a bug in the current ORB, and you might fix it in this PR. The problem is that if you call `measureText` it uses the last set of `context.font` for the measurements.
This means if someone changes the font size, the measurement for the background will be incorrect (it will be calculated on the default font size), but the correct font size will be set only after when the text is being draw with `context.font = label.fontFamily;`

So before doing a `context.measureText` we need to be sure we have a correct font setup, using `context.font = label.fontFamily;`.

Reply:
done